client/jetpack-cloud/sections/partner-portal/issue-license-v2/licenses-form/hooks/use-product-and-plans.tsx

```diff
@@ -12,13 +12,67 @@ import {
 	addSelectedProductSlugs,
 	clearSelectedProductSlugs,
 } from 'calypso/state/partner-portal/products/actions';
+import { APIProductFamilyProduct } from 'calypso/state/partner-portal/types';
+import {
+	PRODUCT_FILTER_ALL,
+	PRODUCT_FILTER_PLANS,
+	PRODUCT_FILTER_PRODUCTS,
+	PRODUCT_FILTER_VAULTPRESS_BACKUP_ADDONS,
+	PRODUCT_FILTER_WOOCOMMERCE_EXTENSIONS,
+} from '../../constants';
 import type { SiteDetails } from '@automattic/data-stores';
 
 type Props = {
 	selectedSite?: SiteDetails | null;
+	selectedProductFilter?: string | null;
+};
+
+const getProductsAndPlansByFilter = (
+	filter: string | null,
+	allProductsAndPlans?: APIProductFamilyProduct[]
+) => {
+	switch ( filter ) {
+		case PRODUCT_FILTER_PRODUCTS:
+			return (
+				allProductsAndPlans?.filter(
+					( { family_slug }: { family_slug: string } ) =>
+						family_slug !== 'jetpack-packs' &&
+						family_slug !== 'jetpack-backup-storage' &&
+						! isWooCommerceProduct( family_slug ) &&
+						! isWpcomHostingProduct( family_slug )
+				) || []
+			);
+		case PRODUCT_FILTER_PLANS:
+			return (
+				allProductsAndPlans?.filter(
+					( { family_slug }: { family_slug: string } ) => family_slug === 'jetpack-packs'
+				) || []
+			);
+
+		case PRODUCT_FILTER_VAULTPRESS_BACKUP_ADDONS:
+			return (
+				allProductsAndPlans
+					?.filter(
+						( { family_slug }: { family_slug: string } ) => family_slug === 'jetpack-backup-storage'
+					)
+					.sort( ( a, b ) => a.product_id - b.product_id ) || []
```

Comment:
Do we sort the results to show the storage options in ascending order (1GB, 10GB, 1TB, etc.)?

Reply:
I believe so, removing this line of code does make the Backup addons 100GB as the first item instead of the 10GB option.